components/selection/selection-observer-mixin.js

```diff
@@ -79,4 +85,9 @@ export const SelectionObserverMixin = superclass => class extends superclass {
 			this.selectionInfo = new SelectionInfo();
 		}
 	}
+
+	_updateProviderFor() {
+		const selectionComponent = this.selectionFor && this.getRootNode().querySelector(`#${cssEscape(this.selectionFor)}`);
```

Comment:
I've left this as `querySelector` from the original logic, but it seems to me that it could be replaced with `getElementById`.

Also I've added the `this.selectionFor &&` check - in the original logic, if `selectionFor` changed from a string to undefined/empty, the querySelector was failing with an error.